### DISCUSSION TITLE
Cert verify fails since last update

### DISCUSSION THREAD

**Discussion by Amply666** [schwabe/ics-openvpn#1387] [Q&A] [answered]
Hallo!!!
Since yesterday, after the last update the program started having problems, in the logs it has the following message:
_error:0A000086:SSL routines::certificate verify failed_

I try to ri download the SSL configuration file but work only in the olds versions (0.7.23) since 0.7.25 same configuration file don't work.

**Comment by tr3027** [comment]
you may have similar issue I had [here](https://github.com/schwabe/ics-openvpn/discussions/1385). In my case [the FAQ](https://ics-openvpn.blinkt.de/FAQ.html#weakmd_title) helped

**Reply by Amply666** [reply]
Thankyou, 
I think the problem is similar, I read FAQ but I can't understand how you fix the problem.
You change somethink in configuration file (*.ovpn) or server side?

**Comment by schwabe** [comment]
From that message alone it is hard to figure out what your exact problem is. Try to provide a more complete log. But yes. It is probably related to stricter checks in OpenSSL 3.0

**Reply by Amply666** [reply]
Yes sure: 
_16:12 EVENT(Error): CERT_VERIFY_FAIL: OpenSSLContext::SSL::read_cleartext: BIO_read failed, cap=2576 status=-1: error:0A000086:SSL routines::certificate verify failed_

I hope this can help you.
Thank you

**Reply by schwabe** [reply]
The lines above that line might contain more information why OpenSSL does not like the certificate.

**Comment by Quikfix73** [comment, marked as answer]
I have the exact same error (refer to my discussion below):

https://github.com/schwabe/ics-openvpn/discussions/1390

The best thing you can do is contact your VPN provider and have them provide you with a newer and stronger certificate in a revised OVPN file.

**Reply by Amply666** [reply]
OK thankyou, is possible in some way to "disable or go back" from new security level?
if I use _OpenVPN for Android_ I have this problem but if I use _OpenVPN_ the same certificate will work properly.

**Reply by schwabe** [reply]
See the FAQ. It explains this.

**Reply by Amply666** [reply]
> If you really want to use old and broken certificates use the custom configuration option tls-cipher "DEFAULT:@SECLEVEL=0" under advanced configuration or as additional line in your imported configuration

I understood this but I find it difficult to find how to do it on the Android APP, I added the line indicated in the configuration file but I have not solved the problem

**Reply by Quikfix73** [reply]
Did you try it without the quotes?  You need to remove any other tls-cipher line that currently exists in your OVPN configuration file and REPLACE it with this below without the quotes:

tls-cipher DEFAULT:@SECLEVEL=0

I tested it and it works fine for me.

In my test OVPN configuration file I had this line below:

tls-cipher TLS-DHE-RSA-WITH-AES-256-CBC-SHA:TLS-DHE-DSS-WITH-AES-256-CBC-SHA:TLS-RSA-WITH-AES-256-CBC-SHA

so I deleted this line above and replaced it with this line below (without the quotes):

tls-cipher DEFAULT:@SECLEVEL=0

and it works and connects fine now but personally I only did this as a test as I would NEVER connect to any VPN service without a STRONG certificate as you become more susceptible to hacking since you are now connecting without any certificate since the Security Level is now ZERO.
